libs/island-ui/core/src/lib/DropdownMenu/useMenuHoverProps.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react'
+import { MenuStateReturn } from 'reakit/Menu'
+
+interface HoverPropsRefState {
+  hoverCount: number
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  hideTimeout?: any
+}
```

Comment:
I went down this rabbit hole once of typing a timeout value, and learned that typescript has support for it with `NodeJS.Timeout`

```ts
hideTimeout?: string | number | NodeJS.Timeout
```

Reply:
Yeah I already saw this. 

Just annoying that typescript has a different type from browsers and to need NodeJS types in our react component library. But that's probably fine? 🤷‍♂️

I'll try it out.

Reply:
Yeah thats true. Someone here mentions using `ReturnType` 
https://stackoverflow.com/a/56239226/9491688

Reply:
Oooh, that's much cleaner!